crates/bevy_text/src/pipeline.rs

```diff
@@ -343,6 +343,31 @@ impl TextPipeline {
             result
         });
 
+        layout_info.section_rects.clear();
+        for run in buffer.layout_runs() {
+            let Some((mut current_section, mut start, mut end)) =
+                run.glyphs.first().map(|g| (g.metadata, g.x, g.x + g.w))
+            else {
+                continue;
+            };
+            for glyph in run.glyphs.iter() {
+                let section_index = glyph.metadata;
+                if section_index != current_section {
+                    layout_info.section_rects.push((
+                        computed.entities[current_section].entity,
+                        Rect::new(start, run.line_top, end, run.line_top + run.line_height),
+                    ));
+                    start = end;
```

Comment:
```suggestion
                    start = glyph.x;
```
It may be more correct/future-proof ? Although I'm not really sure that it can happen in practice to have horizontal space between 2 sections in the same "run".

Reply:
I used `end` to avoid overlaps, but it doesn't seem like either can happen. I'll switch it to `end.min(glyph.x)`, that seems the most robust.